Add CourseEditor model

mikix opened the pull request:
This will tie a course and a user together to determine if they can edit a course.

Also modify the course API view to check for editors or staff state before letting a user modify a course (or organization membership before letting a user create a course).

I left the course run API alone for now - it doesn't yet support editing.

https://openedx.atlassian.net/browse/DISCO-555

jlajoie commented on course_discovery/apps/api/permissions.py:
```diff
@@ -11,11 +11,24 @@ def has_permission(self, request, view):
+    def has_object_permission(self, request, view, obj):
+        if request.user.is_staff:  # staff users are always editors
+            return True
+        elif request.method in SAFE_METHODS:
```
Is SAFE_METHODS just like GET requests?

mikix replied:
Yeah, exactly: ('GET', 'HEAD', 'OPTIONS')

mikix commented on course_discovery/apps/api/permissions.py:
```diff
@@ -11,11 +11,24 @@ def has_permission(self, request, view):
+        elif request.method in SAFE_METHODS:
+            return True
+        else:
+            return request.user.courses_edited.filter(course=obj).exists()
```
`courses_edited` is maybe a bad name I gave it? I meant "courses for which this user is an editor" -- is there a less confusing way of saying that?

You are right that I didn't add any logic around handling when someone is removed from an org, removing all their editor access. I think that would be something good to do... Do you know a non-painful way of adding that logic?

mikix commented:
Oh, I also forgot one aspect of that ADR I wrote - if a course has no editors for whatever reason, everyone in its authoring orgs is an editor... will fix

mikix commented:
OK, updated permission logic and added new tests -- now we allow access if no other valid editors and we check group membership every access, rather than assuming if you're an editor, you're always allowed in.
